Lib/doctest.py

```diff
@@ -1627,6 +1637,15 @@ def merge(self, other):
             d[name] = (failures, tries, skips)
 
 
+def _n_items(items: list) -> str:
```

Comment:
You could consider adding
```suggestion
def pl(n: int) -> str:
    "  Helper to add 's' when  plural."
    return "s" if n != 1 else ""


def _n_items(items: list) -> str:
```
This would be used 3 times above and once below.  "pl" condenses "'s' if is_plural".

Reply:
I also considered letting `_n_items()` take an `int` as well, and use `isinstance` to decide whether to call `len()`.

We'd also need to pass in the unit name. For example:

```python
from collections.abc import Sized


def _n_items(n: int | Sized, unit: str = "item") -> str:
    """
    Helper to pluralise the number of items.
    """
    if isinstance(n, Sized):
        n = len(n)

    s = "" if n == 1 else "s"
    return f"{n} {unit}{s}"
```

Then call it like:

```python
# Existing, no change:
print(f"{_n_items(failed)} had failures:")

# New ones:
print(f"{_n_items(total_tries, "test")} in {_n_items(self._stats)}.")
...
msg = f"***Test Failed*** {_n_items(total_failures, "failure")}"
...
msg = f"{msg} and skipped {_n_items(total_skips, "test")}"
```

What do you think?

Reply:
I think that it is overengineering. In distant future we may add internationalization using `gettext`, and it uses completely different approach.